js/interact.js

```diff
@@ -195,15 +195,44 @@ function uiInteractLegend (metricQInstance, evtObj) {
     }
   }
   const timeString = posDate.toLocaleString()
-  myCtx.fillText(timeString, curPosOnCanvas[0] + 10, 40 - 20)
+  const offsetTop = 30
+  const verticalDiff = 20
+  const borderPadding = 10
+  const offsetMid = 10
+
+  myCtx.textBaseline = 'middle'
+  drawHoverDate(myCtx, timeString, metricQInstance, curPosOnCanvas[0], maxTextWidth, offsetTop, offsetMid, verticalDiff, borderPadding)
+  drawHoverText(myCtx, metricsArray, metricQInstance, curPosOnCanvas[0], minTextWidth, maxTextWidth, offsetTop, offsetMid, verticalDiff, borderPadding)
+}
+
+function drawHoverDate (myCtx, timeString, metricQInstance, curPosOnCanvas, maxTextWidth, offsetTop, offsetMid, verticalDiff, borderPadding) {
```

Comment:
We should not have a variable with the same name and different contents (curPosOnCanvas sometimes being a array, sometimes being the x-Value). E.g. rename to `curXPosOnCanvas`.